conformant = "Partial"
notes = """
Does not report error for a forward reference that is not enclosed in quotes.
Does not report error for use of quoted type with "|" operator (runtime error).
Does not reject f-string in quoted type annotation.
Incorrectly generates error for quoted type defined in class scope.
Does not generate error for unquoted type defined in class scope.
Does not treat triple-quoted forward reference annotation as implicitly parenthesized.
"""
output = """
annotations_forward_refs.py:41:8 Invalid type [31]: Expression `eval("".join(map(chr, [105, 110, 116])))` is not a valid type.
annotations_forward_refs.py:42:8 Invalid type [31]: Expression `"[int, str]"` is not a valid type.
annotations_forward_refs.py:43:8 Invalid type [31]: Expression `"(int, str)"` is not a valid type.
annotations_forward_refs.py:44:8 Undefined or invalid type [11]: Annotation `comprehension(int for generators(generator($target$i in range(1) if )))` is not defined as a type.
annotations_forward_refs.py:45:8 Invalid type [31]: Expression `"{  }"` is not a valid type.
annotations_forward_refs.py:46:8 Invalid type [31]: Expression `lambda () (int)()` is not a valid type.
annotations_forward_refs.py:47:8 Invalid type [31]: Expression `[int][0]` is not a valid type.
annotations_forward_refs.py:48:8 Invalid type [31]: Expression `"int if 1 < 3 else str"` is not a valid type.
annotations_forward_refs.py:49:8 Undefined or invalid type [11]: Annotation `var1` is not defined as a type.
annotations_forward_refs.py:50:9 Invalid type [31]: Expression `"True"` is not a valid type.
annotations_forward_refs.py:51:9 Invalid type [31]: Expression `"1"` is not a valid type.
annotations_forward_refs.py:52:9 Invalid type [31]: Expression `"-1"` is not a valid type.
annotations_forward_refs.py:53:9 Invalid type [31]: Expression `"int or str"` is not a valid type.
annotations_forward_refs.py:55:9 Undefined or invalid type [11]: Annotation `types` is not defined as a type.
annotations_forward_refs.py:80:12 Undefined or invalid type [11]: Annotation `ClassF` is not defined as a type.
annotations_forward_refs.py:87:7 Undefined or invalid type [11]: Annotation `ClassD.int` is not defined as a type.
annotations_forward_refs.py:96:0 Assert type [70]: Expected `int` but got `unknown`.
annotations_forward_refs.py:103:7 Undefined or invalid type [11]: Annotation `
"""
conformance_automated = "Fail"
errors_diff = """
Line 22: Expected 1 errors
Line 23: Expected 1 errors
Line 24: Expected 1 errors
Line 25: Expected 1 errors
Line 54: Expected 1 errors
Line 66: Expected 1 errors
Line 89: Expected 1 errors
Line 87: Unexpected errors ['annotations_forward_refs.py:87:7 Undefined or invalid type [11]: Annotation `ClassD.int` is not defined as a type.']
Line 96: Unexpected errors ['annotations_forward_refs.py:96:0 Assert type [70]: Expected `int` but got `unknown`.']
Line 103: Unexpected errors ['annotations_forward_refs.py:103:7 Undefined or invalid type [11]: Annotation `']
"""
